help/help.c

```diff
@@ -705,20 +711,17 @@ int help_doclist_init(struct Vector *DocList)
  */
 static int help_doclist_parse(struct Mailbox *m)
 {
-  ///< all valid help documents within $help_doc_dir folder
+  // all valid help documents within $help_doc_dir folder
   // Local variable to be saved in Mailbox
-  struct Vector *DocList = vector_new(sizeof(struct Email));
+  struct Vector *DocList = vector_new(sizeof(struct Email *));
 
   if ((help_doclist_init(DocList) != 0) || (DocList->size == 0))
     return -1;
 
-  m->mdata = vector_clone(DocList, true, help_doc_clone);
-  m->free_mdata = help_doclist_free;
-  m->emails = (struct Email **) ((struct Vector *) m->mdata)->data;
+  m->emails = (struct Email **) DocList->data;
   m->msg_count = m->email_max = DocList->size;
   mutt_mem_realloc(&m->v2r, sizeof(int) * m->email_max);
-
-  mutt_make_label_hash(m);
+  FREE(&DocList);
```

Comment:
At this point, the ownership of the array changes from `DocList` to the `Mailbox`.
Free the `DocList` as it's now empty.